diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostFunctionContext.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostFunctionContext.swift
index bc89b9c..7781dd2 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostFunctionContext.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostFunctionContext.swift
@@ -2,7 +2,7 @@
  Context that captures Swift values to pass them to JSI host function as an unmanaged pointer for interoperability with C++.
  */
 internal final class HostFunctionContext: Sendable {
-  weak let runtime: JavaScriptRuntime?
+  nonisolated(unsafe) weak var runtime: JavaScriptRuntime?
   let name: String?
   let call: JavaScriptRuntime.SyncFunctionClosure
 
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostObjectContext.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostObjectContext.swift
index f61fcda..47f19f2 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostObjectContext.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Contexts/HostObjectContext.swift
@@ -7,7 +7,7 @@ internal final class HostObjectContext: Sendable {
   typealias PropertyNamesGetter = @JavaScriptActor () -> [String]
   typealias Deallocator = @JavaScriptActor () -> Void
 
-  weak let runtime: JavaScriptRuntime?
+  nonisolated(unsafe) weak var runtime: JavaScriptRuntime?
   let get: Getter
   let set: Setter?
   let getPropertyNames: PropertyNamesGetter
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptActor.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptActor.swift
index 699c59a..b3be83c 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptActor.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptActor.swift
@@ -93,7 +93,7 @@ internal class JavaScriptExecutor: SerialExecutor, @unchecked Sendable {
  An actor that is dedicated for the specific runtime.
  */
 internal actor JavaScriptRuntimeActor {
-  private weak let runtime: JavaScriptRuntime?
+  private weak var runtime: JavaScriptRuntime?
   nonisolated private let executor: JavaScriptExecutor
 
   init(runtime: JavaScriptRuntime) {
@@ -114,7 +114,7 @@ internal actor JavaScriptRuntimeActor {
  Serial executor dedicated for the specific runtime.
  */
 internal final class JavaScriptRuntimeExecutor: JavaScriptExecutor, @unchecked Sendable {
-  private weak let runtime: JavaScriptRuntime?
+  private weak var runtime: JavaScriptRuntime?
 
   init(runtime: JavaScriptRuntime) {
     self.runtime = runtime
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptPropNameID.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptPropNameID.swift
index b497d4f..4390c5b 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptPropNameID.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/JavaScriptPropNameID.swift
@@ -4,7 +4,7 @@ internal import jsi
  Represents something that can be a JS property key.
  */
 public final class JavaScriptPropNameID: JavaScriptType {
-  private weak let runtime: JavaScriptRuntime?
+  nonisolated(unsafe) private weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.PropNameID
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArray.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArray.swift
index 12acf78..98c1fff 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArray.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArray.swift
@@ -7,7 +7,7 @@ internal import ExpoModulesJSI_Cxx
  to the underlying JavaScript array and provides Swift-friendly APIs for common array operations.
  */
 public struct JavaScriptArray: JavaScriptType, ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.Array
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArrayBuffer.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArrayBuffer.swift
index 7289522..e948877 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArrayBuffer.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptArrayBuffer.swift
@@ -8,7 +8,7 @@ internal import ExpoModulesJSI_Cxx
  underlying raw data pointer and size of the buffer.
  */
 public struct JavaScriptArrayBuffer: ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.ArrayBuffer
 
   internal init(_ runtime: JavaScriptRuntime, _ arrayBuffer: consuming facebook.jsi.ArrayBuffer) {
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptBigInt.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptBigInt.swift
index c9e68e5..550c3dd 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptBigInt.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptBigInt.swift
@@ -39,7 +39,7 @@ internal import jsi
  You must explicitly convert between them.
  */
 public struct JavaScriptBigInt: JavaScriptType, Sendable, ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal var pointee: facebook.jsi.BigInt
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptError.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptError.swift
index 79a2fbb..dc6d266 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptError.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptError.swift
@@ -2,7 +2,7 @@ internal import jsi
 internal import ExpoModulesJSI_Cxx
 
 public struct JavaScriptError: JavaScriptType, ~Copyable {
-  private weak let runtime: JavaScriptRuntime?
+  private weak var runtime: JavaScriptRuntime?
   nonisolated(unsafe) private let pointee: facebook.jsi.JSError
 
   internal init(_ runtime: JavaScriptRuntime, _ error: facebook.jsi.JSError) {
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptFunction.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptFunction.swift
index 53c2445..9c14efd 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptFunction.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptFunction.swift
@@ -4,7 +4,7 @@ internal import jsi
 internal import ExpoModulesJSI_Cxx
 
 public struct JavaScriptFunction: JavaScriptType, ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.Function
 
   internal init(_ runtime: JavaScriptRuntime, _ pointee: consuming facebook.jsi.Function) {
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptObject.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptObject.swift
index 57bb662..9d7d22d 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptObject.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptObject.swift
@@ -8,7 +8,7 @@ internal import ExpoModulesJSI_Cxx
  supporting property access, modification, enumeration, prototype manipulation, and function calling.
  */
 public struct JavaScriptObject: JavaScriptType, Sendable, ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal var pointee: facebook.jsi.Object
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptPromise.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptPromise.swift
index 94f10ed..c3b6b69 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptPromise.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptPromise.swift
@@ -12,7 +12,7 @@ internal import ExpoModulesJSI_Cxx
 public struct JavaScriptPromise: JavaScriptType, ~Copyable {
   private typealias PromiseContinuation = CheckedContinuation<JavaScriptValue.Ref, any Error>
 
-  private weak let runtime: JavaScriptRuntime?
+  private weak var runtime: JavaScriptRuntime?
   private let object: JavaScriptObject
   private let deferredPromise = DeferredPromise()
 
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptTypedArray.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptTypedArray.swift
index c8caee5..7b6d481 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptTypedArray.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptTypedArray.swift
@@ -4,7 +4,7 @@ internal import jsi
 internal import ExpoModulesJSI_Cxx
 
 public struct JavaScriptTypedArray: ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.Object
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptValue.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptValue.swift
index ba58f9e..efcc970 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptValue.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptValue.swift
@@ -9,7 +9,7 @@ internal import ExpoModulesJSI_Cxx
  and stored in containers that don't support non-copyable types etc.
  */
 public final class JavaScriptValue: JavaScriptType, Equatable, Escapable, Error {
-  internal weak let runtime: JavaScriptRuntime?
+  nonisolated(unsafe) internal weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.Value
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptWeakObject.swift b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptWeakObject.swift
index 2907dad..d4f9411 100644
--- a/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptWeakObject.swift
+++ b/node_modules/expo-modules-jsi/apple/Sources/ExpoModulesJSI/Runtime/Values/JavaScriptWeakObject.swift
@@ -7,7 +7,7 @@ internal import jsi
  the object is eligible for GC. Method names are inspired by C++ `std::weak_ptr`.
  */
 public struct JavaScriptWeakObject: JavaScriptType, ~Copyable {
-  internal weak let runtime: JavaScriptRuntime?
+  internal weak var runtime: JavaScriptRuntime?
   internal let pointee: facebook.jsi.WeakObject
 
   /**
diff --git a/node_modules/expo-modules-jsi/apple/scripts/build-xcframework.sh b/node_modules/expo-modules-jsi/apple/scripts/build-xcframework.sh
index 4d98726..167f52e 100755
--- a/node_modules/expo-modules-jsi/apple/scripts/build-xcframework.sh
+++ b/node_modules/expo-modules-jsi/apple/scripts/build-xcframework.sh
@@ -153 +153,2 @@ build_slice() {
+  rm -rf "$DERIVED_DATA_PATH"
   rm -rf "$BUILD_PRODUCTS_PATH"
